kernel/src/scan/log_replay.rs

```diff
@@ -103,7 +140,21 @@ impl AddRemoveDedupVisitor<'_> {
 
         // Process both adds and removes, but only return not already-seen adds
         let file_key = FileActionKey::new(path, dv_unique_id);
-        Ok(!self.check_and_record_seen(file_key) && is_add)
+        let have_seen = self.check_and_record_seen(file_key);
+        if !is_add || have_seen {
+            return Ok(false);
+        }
```

Comment:
This seems like a dangerous change (because somebody trying to "optimize" the code might produce control flow that skips non-adds without checking them first). Now that you no longer need the `have_seen` multiple times, can we partially revert so it matches the code comment at L142 again?
```suggestion
        if self.check_and_record_seen(file_key) || !is_add {
            return Ok(false);
        }
```

Reply:
Actually, we should probably update the comment to match the new code:
```rust
// Check both adds and removes (skipping already-seen), but only transform and return adds